fe/src/main/java/org/apache/doris/clone/TabletScanner.java

```diff
@@ -0,0 +1,181 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.clone;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Sets;
+
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.MaterializedIndex;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.Partition;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.catalog.Tablet;
+import org.apache.doris.catalog.Tablet.TabletStatus;
+import org.apache.doris.common.Pair;
+import org.apache.doris.common.util.Daemon;
+import org.apache.doris.system.SystemInfoService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Set;
+
+/*
+ * This scanner is responsible for scanning all unhealthy tablets.
+ * It does not responsible for any scheduler of tablet repairing or balance
+ */
+public class TabletScanner extends Daemon {
+    private static final Logger LOG = LogManager.getLogger(TabletScanner.class);
+
+    private static long INTERVAL_MS = 60 * 1000L; // 1min
+
+    private Catalog catalog;
+    private SystemInfoService infoService;
+    private TabletFactory tabletFactory;
+
+    // db id -> (tbl id -> partition id)
+    // priority of replicas of partitions in this table will be set to VERY_HIGH if not healthy
+    private com.google.common.collect.Table<Long, Long, Set<Long>> priors = HashBasedTable.create();
+
+    public TabletScanner() {
+        super("tablet scanner", INTERVAL_MS);
+        catalog = Catalog.getCurrentCatalog();
+        infoService = Catalog.getCurrentSystemInfo();
+        tabletFactory = new TabletFactory();
+    }
+
+    public void addPriors(long dbId, long tblId, List<Long> partitionIds) {
```

Comment:
"Prior" is adjective

Reply:
I just want use an abbreviation, which should be prio. I will fix it.